src/handlers.cpp

```diff
@@ -1578,29 +1640,120 @@ void ImpuRegDataTask::on_sar_response(Diameter::Message& rsp)
       {
         TRC_DEBUG("Getting User-Data from SAA for cache");
         saa.user_data(_xml);
-        put_in_cache();
+        caching = true;
```

Comment:
This wasn't quite what I meant.  See https://github.com/Metaswitch/homestead/commit/c1536ce6465a33e4745b5e91ad2b59e40e669478 for what I was suggesting - basically, all the caching logic becomes its own unit, so the processing in this function consists of discrete blocks:

*   Parse the response/extract information.
*   Update the cache.
*   Send the response (or don't if there's a pending cache operation).

This feels more maintainable.